src/drivers/actuators/vertiq_io/vertiq_client_manager.cpp

```diff
@@ -0,0 +1,208 @@
+	_voltage_max_entry.ConfigureStruct(param_find("MAX_VOLTS"), &(_prop_input_parser_client->volts_max_));
+	_control_mode_entry.ConfigureStruct(param_find("CONTROL_MODE"), &(_prop_input_parser_client->mode_));
+	_motor_direction_entry.ConfigureStruct(param_find("VERTIQ_MOTOR_DIR"), &(_prop_input_parser_client->sign_));
+	_fc_direction_entry.ConfigureStruct(param_find("VERTIQ_FC_DIR"), &(_prop_input_parser_client->flip_negative_));
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	_throttle_cvi_entry.ConfigureStruct(param_find("THROTTLE_CVI"), &(_ifci_client->throttle_cvi_));
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	_pulsing_voltage_mode_entry.ConfigureStruct(param_find("PULSE_VOLT_MODE"),
+			&(_pulsing_rectangular_input_parser_client->pulsing_voltage_mode_));
+	_x_cvi_entry.ConfigureStruct(param_find("X_CVI"), &(_ifci_client->x_cvi_));
+	_y_cvi_entry.ConfigureStruct(param_find("Y_CVI"), &(_ifci_client->y_cvi_));
+	_pulse_zero_angle_entry.ConfigureStruct(param_find("ZERO_ANGLE"), &(_voltage_superposition_client->zero_angle_));
+	_pulse_velo_cutoff_entry.ConfigureStruct(param_find("VELOCITY_CUTOFF"),
+			&(_voltage_superposition_client->velocity_cutoff_));
+	_pulse_torque_offset_entry.ConfigureStruct(param_find("TORQUE_OFF_ANGLE"),
+			&(_voltage_superposition_client->propeller_torque_offset_angle_));
+	_pulse_volt_limit_entry.ConfigureStruct(param_find("PULSE_VOLT_LIM"),
+						&(_pulsing_rectangular_input_parser_client->pulsing_voltage_limit_));
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+}
+
+uint8_t VertiqClientManager::GetObjectIdNow()
+{
+	return _object_id_now;
+}
+
+void VertiqClientManager::UpdateClientsToNewObjId(uint8_t new_object_id)
+{
+
+	_object_id_now = new_object_id;
+
+	delete _prop_input_parser_client;
+	_prop_input_parser_client = new EscPropellerInputParserClient(new_object_id);//&prop_input_parser;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	delete _ifci_client;
+	_ifci_client = new IQUartFlightControllerInterfaceClient(new_object_id);
+#endif
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	delete _voltage_superposition_client;
+	_voltage_superposition_client = new VoltageSuperPositionClient(new_object_id);
+
+	delete _pulsing_rectangular_input_parser_client;
+	_pulsing_rectangular_input_parser_client = new PulsingRectangularInputParserClient(new_object_id);
+#endif
+}
+
+void VertiqClientManager::HandleClientCommunication()
+{
+	//Called periodically in the main loop to handle all communications not handled direclty by
+	//parameter setting
+	//Update our serial tx before we take in the rx
+	_serial_interface->process_serial_tx();
+
+	//Update our serial rx
+	_serial_interface->process_serial_rx(&_motor_interface, _client_array);
+}
+
+IFCI *VertiqClientManager::GetMotorInterface()
+{
+	return &_motor_interface;
+}
+
+uint8_t VertiqClientManager::GetNumberOfClients()
+{
+	return _clients_in_use;
+}
+
+void VertiqClientManager::SendSetForceArm()
```

Comment:
here are some helpers that the control loop can use. just call the function, don't worry about entries and values.